### PR TITLE
Update workflows and readme to current versions from IUC

### PR DESCRIPTION
**Pull request by richard-burhans**
I updated the {ci,pr,slash}.yaml workflows and README.md to the current versions in the IUC repo.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -7,12 +7,12 @@ on:
     types: [run-all-tool-tests-command]
 env:
   GALAXY_FORK: galaxyproject
-  GALAXY_BRANCH: release_21.05
+  GALAXY_BRANCH: release_24.0
```
Comment (bernt-matthias) [inline .github/workflows/ci.yaml:10]: Maybe we go directly to 24.1 (release is expected before GCC). Same in the PR workflow.

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -159,57 +187,64 @@ jobs:
...
-              write(paste(paste(rel_path, l$line_number, l$column_number, sep=":"), l$message), stderr())
-              write(paste(paste(rel_path, l$line_number, l$column_number, sep=":"), l$message), "rlint_report.txt", append=TRUE)
-            }
-          }
-        }
-        quit(status = status)
-      shell: Rscript {0}
-    - uses: actions/upload-artifact@v2
+        set -eo pipefail
+        echo '${{ needs.setup.outputs.repository-list }}' | xargs -d '\n' -n 1 ./.github/styler.R --dry off
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:212]: You will need to add `./.github/styler.R`

**Review by bernt-matthias — COMMENTED** [review]
Thanks for doing this. We need to add also https://github.com/galaxyproject/tools-iuc/blob/main/.github/workflows/pr_without_tool_change.yaml

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (richard-burhans) [reply .github/workflows/ci.yaml]: I agree.  I've updated the version to 24.1

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**